datalad/support/gitrepo.py

```diff
@@ -719,6 +719,24 @@ def __init__(self, path, url=None, runner=None, create=True,
         self._fake_dates_enabled = None
 
     def _create_empty_repo(self, path, **kwargs):
+        if op.lexists(path):
+            # Verify that we are not trying to initialize a new git repository
+            # under a directory some files of which are already tracked by git
+            # use case: https://github.com/datalad/datalad/issues/3068
+            try:
+                stdout, _ = self._git_custom_command(
+                    None, ['git', 'ls-files'], cwd=path, expect_fail=True
```

Comment:
Doubt it matters, but using `git ls-tree HEAD` (no `-r`) would avoid listing all the contents of a deep tree with many files under directories.

Reply:
hm... 
`ls-tree` (without HEAD) even fails to run in a top directory of a valid (even without any file in it, just freshly initialized like in #3139) git repo:
```shell
(git-annex)hopa:~/datalad/openfmri/ds000001[master]
$> git ls-tree ; echo $?
usage: git ls-tree [<options>] <tree-ish> [<path>...]

    -d                    only show trees
    -r                    recurse into subtrees
    -t                    show trees when recursing
    -z                    terminate entries with NUL byte
    -l, --long            include object size
    --name-only           list only filenames
    --name-status         list only filenames
    --full-name           use full path names
    --full-tree           list entire tree; not just current directory (implies --full-name)
    --abbrev[=<n>]        use <n> digits to display SHA-1s

129
```
and that is when we also do not want to re-init probably.
in an empty one, `ls-tree HEAD` exits with 128 exit code.

and now I realized you were talking about not only adding `HEAD` but changing to `ls-tree` from `ls-files`.  

FTR, `ls-files` with and without HEAD, returns nothing and doesn't error out in an empty repo.

So, indeed `ls-tree HEAD` in ds000001 takes 0.003 sec, `ls-files` takes  0.007.  Let me switch ;)  (FTR: `git -c pager.config=0 config -l -z --show-origin` takes also 0.003 sec)

Reply:
hm -- with `ls-tree HEAD` we are not catching then the case where files are already in the index but not yet committed...  so we would not be catching that.  Given that running ls-files in an empty subdirectory takes on that ds000001 only 0.003-0.005 sec, I would then prefer to keep it as is.

I spotted that there is also `--deleted` option for `ls-files` but add/commit/rm , create another new one, resulted in an interesting output of git status:
```shell
$> time git ls-files --deleted
git ls-files --deleted  0.00s user 0.00s system 88% cpu 0.005 total
```
so `--deleted` didn't give me the file which is still known to git but deleted/not yet committed,

Do you know a way for those to show up?

Reply:
> so `--deleted` didn't give me the file which is still known to git but deleted/not yet committed,
> 
> Do you know a way for those to show up?

AFAIU `git ls-files` won't report on it because it is only concerned about files in the working tree or index.

Reply:
Right, but isn't --deleted exactly for this use case? I am now just curious what it's purpose

Reply:
> Right, but isn't --deleted exactly for this use case?

I think it's to distinguish files that are in the index but not the working tree.